src/strnum.F90

```diff
@@ -0,0 +1,73 @@
+!> @file
+!> @brief Decode an integer from a character string.
+!>
+!> @author J. Woollen @date 1994-01-06
+
+!> This subroutine decodes an integer from a character string.  The
+!> string may contain leading or trailing blanks, but otherwise should
+!> contain only digits and an (optional) leading sign ('+' or '-')
+!> character. 
+!>
+!> If the string contains all blank characters, then num is returned
+!> with a value of 0.
+!>
+!> @param[in]  str -- character*(*): String
+!> @param[out] num -- integer: Value decoded from str
+!> @param[out] iret -- return code:
+!>   0 = success
+!>  -1 = string contained one or more illegal characters
+!>
+!> @author J. Woollen @date 1994-01-06
+recursive subroutine strnum(str,num,iret)
+  use modv_im8b
+
+  implicit none
+
+  character*(*), intent(in) :: str
+
+  integer, intent(out) ::  num, iret
+
+  character str2*40, fmt*8
+
+  integer lens
+
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+
+! Check for I8 integers.
+
+  if (im8b) then
+     im8b = .false.
+
+     call strnum ( str, num, iret )
+     call x48 ( num, num, 1 )
+     call x48 ( iret, iret, 1 )
+
+     im8b = .true.
+     return
+  end if
+
+  iret = 0
+
+! Check for blank input string.
+
+  if ( str .eq. ' ' ) then
+    num = 0
+    return
+  end if
+
+! Verify that the string contains all legal characters.
+
+  call strsuc ( str, str2, lens )
```

Comment:
Why strsuc? Can't trim() do this already?

Reply:
If you look at my new strsuc in this PR (where I've completely rewritten it and migrated it to F90), you'll see that it now does nothing more than call the intrinsic functions adjustl and len_trim ;-)

It's true that, after having done that, I could have just done away with strsuc altogether and just called adjustl and len_trim directly from within strnum.F90.  But then I'd have to do likewise in a bunch of different subprograms in the library which also call strsuc, so I decided (again, just IMHO) to do it this way.